tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/greater-less-than-operation.bal

```diff
@@ -660,3 +660,718 @@ function testUnorderedTypeComparison8() {
+
+    boolean x0 = a0 > b0;
+    boolean x1 = b0 > a0;
+    test:assertFalse(x0);
+    test:assertFalse(x1);
+
+    boolean x2 = a0 >= b0;
+    boolean x3 = b0 >= a0;
+    test:assertFalse(x2);
+    test:assertFalse(x3);
+
+    boolean x4 = a0 < b0;
+    boolean x5 = b0 < a0;
+    test:assertFalse(x4);
+    test:assertFalse(x5);
+
+    boolean x6 = a0 <= b0;
+    boolean x7 = b0 <= a0;
+    test:assertFalse(x6);
+    test:assertFalse(x7);
+
+    boolean x8 = a1 > b0;
+    boolean x9 = b0 > a1;
+    test:assertFalse(x8);
+    test:assertFalse(x9);
+
+    boolean x10 = a1 >= b0;
+    boolean x11 = b0 >= a1;
+    test:assertFalse(x10);
+    test:assertFalse(x11);
+
+    boolean x12 = a1 < b0;
+    boolean x13 = b0 < a1;
+    test:assertFalse(x12);
+    test:assertFalse(x13);
+
+    boolean x14 = a1 <= b0;
+    boolean x15 = b0 <= a1;
+    test:assertFalse(x14);
+    test:assertFalse(x15);
+
+    boolean x16 = a1 < a0;
+    boolean x17 = a0 <= a1;
+    test:assertFalse(x16);
+    test:assertTrue(x17);
+
+    boolean x18 = a1 > a0;
+    boolean x19 = a0 >= a1;
+    test:assertFalse(x18);
+    test:assertTrue(x19);
+}
+
+function testUnorderedTypeComparison18() {
+    string:Char? a0 = ();
+    string:Char b0 = "a";
+
+    boolean x0 = a0 > b0;
+    boolean x1 = b0 > a0;
+    test:assertFalse(x0);
+    test:assertFalse(x1);
+
+    boolean x2 = a0 >= b0;
+    boolean x3 = b0 >= a0;
+    test:assertFalse(x2);
+    test:assertFalse(x3);
+
+    boolean x4 = a0 < b0;
+    boolean x5 = b0 < a0;
+    test:assertFalse(x4);
+    test:assertFalse(x5);
+
+    boolean x6 = a0 <= b0;
+    boolean x7 = b0 <= a0;
+    test:assertFalse(x6);
+    test:assertFalse(x7);
+}
+
+function testUnorderedTypeComparison19() {
+    string? a0 = ();
+    string:Char b0 = "a";
+
+    boolean x0 = a0 > b0;
+    boolean x1 = b0 > a0;
+    test:assertFalse(x0);
+    test:assertFalse(x1);
+
+    boolean x2 = a0 >= b0;
+    boolean x3 = b0 >= a0;
+    test:assertFalse(x2);
+    test:assertFalse(x3);
+
+    boolean x4 = a0 < b0;
+    boolean x5 = b0 < a0;
+    test:assertFalse(x4);
+    test:assertFalse(x5);
+
+    boolean x6 = a0 <= b0;
+    boolean x7 = b0 <= a0;
+    test:assertFalse(x6);
+    test:assertFalse(x7);
+}
+
+type Type1 [boolean?, int, decimal?]; // ordered type -> [boolean?, int, decimal?]
+type Type2 [boolean?, int?]; // ordered type -> [boolean?, int?]
+// for Type1 and Type2, ordered type can be taken as [boolean?, int?, decimal?...]
+
+function testUnorderedTypeComparison20() {
+    Type1 a = [(), 1];
+    Type2 b = [()];
+
+    boolean x0 = a > b;
+    boolean x1 = b > a;
+    test:assertFalse(x0);
+    test:assertFalse(x1);
+
+    boolean x2 = a >= b;
+    boolean x3 = b >= a;
+    test:assertFalse(x2);
+    test:assertFalse(x3);
+
+    boolean x4 = a < b;
+    boolean x5 = b < a;
+    test:assertFalse(x4);
+    test:assertFalse(x5);
+
+    boolean x6 = a <= b;
+    boolean x7 = b <= a;
+    test:assertFalse(x6);
+    test:assertFalse(x7);
```

Comment:
Shall we add some assertTrue cases like,
```ballerina
    Type1 ty1 = [true, 3];
    Type2 ty2 = [true, 4];
    io:println(ty1 < ty2);
```

Reply:
Please check other places too.

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/36191/commits/08db8afd98dbcf7398a740d7e11aaed95e039972.